File: py/tests/local/test_stream_chat.py
"""
Stream chat test — token streaming from AiAgent.
Usage: python tests/local/test_stream_chat.py
"""
import os
import sys
import json
from imbrace import ImbraceClient

ACCESS_TOKEN = os.environ.get("IMBRACE_ACCESS_TOKEN", "acc_c10f483f-0e45-4966-a275-2fb0356365e7")
GATEWAY      = os.environ.get("IMBRACE_GATEWAY_URL",  "https://app-gatewayv2.imbrace.co")
ASSISTANT_ID = os.environ.get("IMBRACE_ASSISTANT_ID", "a5ffe364-c136-40a0-aa49-84866e4d8485")
ORG_ID       = os.environ.get("IMBRACE_ORG_ID",       "org_6d4ae4f2-f75c-4324-9269-c3fec12078cc")

client = ImbraceClient(access_token=ACCESS_TOKEN, gateway=GATEWAY)

def test_stream():
    print("Testing stream_chat (token streaming)...")
    
    resp = client.ai_agent.stream_chat({
        "assistant_id": ASSISTANT_ID,
        "organization_id": ORG_ID,
        "messages": [{"role": "user", "content": "Say hello in one sentence."}],
    })

    print(f"Status: {resp.status_code}")
    
    token_count = 0
    for line in resp.iter_lines():
        if line:
            line_str = line.decode("utf-8")
            if line_str.startswith("data: ") and line_str != "data: [DONE]":
                try:
                    event = json.loads(line_str[6:])
                    if event.get("type") == "text-delta":
                        sys.stdout.write(event.get("delta", ""))
                        sys.stdout.flush()
                        token_count += 1
                except:
                    pass
    
    print(f"\n\n✓ {token_count} text-delta chunks received")

if __name__ == "__main__":
    test_stream()

File: ts/tests/local/test-stream-chat.mjs
import { ImbraceClient } from '@imbrace/sdk'

const ACCESS_TOKEN = 'acc_c10f483f-0e45-4966-a275-2fb0356365e7'
const GATEWAY      = 'https://app-gatewayv2.imbrace.co'
const ASSISTANT_ID = 'a5ffe364-c136-40a0-aa49-84866e4d8485'
const ORG_ID       = 'org_6d4ae4f2-f75c-4324-9269-c3fec12078cc'

const client = new ImbraceClient({ accessToken: ACCESS_TOKEN, gateway: GATEWAY })

console.log('Testing streamChat (token streaming)...')
// user_id omitted — SDK fetches it automatically via chat-client/auth/user
const resp = await client.aiAgent.streamChat({
  assistant_id: ASSISTANT_ID,
  organization_id: ORG_ID,
  messages: [{ role: 'user', content: 'Say hello in one sentence.' }],
})

console.log('Status:', resp.status)
let tokenCount = 0
for await (const chunk of resp.body) {
  const text = new TextDecoder().decode(chunk)
  for (const line of text.split('\n')) {
    if (line.startsWith('data: ') && line !== 'data: [DONE]') {
      try {
        const event = JSON.parse(line.slice(6))
        if (event.type === 'text-delta') {
          process.stdout.write(event.delta)
          tokenCount++
        }
      } catch {}
    }
  }
}
console.log(`\n\n✓ ${tokenCount} text-delta chunks received`)
